scripts/Dockerfile.race.build

```diff
@@ -1,6 +1,6 @@
 # syntax = docker/dockerfile:experimental
 
-FROM registry.ddbuild.io/images/mirror/golang:1.23.0 as builder
+FROM golang:1.22 as builder
```

Comment:
Did we mean to bump this from 1.23 to 1.22?

Reply:
I reverted the merge commit from [#633](https://github.com/DataDog/datadog-lambda-extension/pull/633/files) and filtered out all of the changes except those made to the Dockerfiles in `scripts`. It looks like as part of that change go was bumped from 1.22 to 1.23.

I think if we want to bump go for these Dockerfiles we can do it in a separate PR. This change is really just to get the Github Workflows using these images back to the last stable version.